### PR TITLE
[wallet-ext] update receipt page to show information about gas sponsors

### PR DESCRIPTION
**Pull request by williamrobertson13**
## Description 
This change updates the wallet receipt pages to show extra information for sponsored transactions. This isn't totally up to spec because we don't have enough granularity to identify error messages (ideally we should show "Out of funds" text for failed sponsored transactions where the sponsor ran out of gas), but I'll work with @mystie711 on this as a follow-up.

Failed transaction with error message:
Pasted Graphic 6

Failed sponsored transaction (ignore the "Successful" text - that was for testing):
Pasted Graphic 2

Successful transaction:
Pasted Graphic 5

Successful sponsored transaction:
Pasted Graphic 4

## Test Plan 
- Manual testing (failed transaction w/ and w/o error, successful NFT mint txn, sent txn, received txn, etc.)

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [X] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes
N/A

### CODE REVIEW

**Comment by mystie711** [comment]
Looks good in general Will. Quick question...is the "ERROR" text in this screenshot supposed to be a placeholder for the error message?
image

If so, I would recommend not capitalizing the color and using the error message color.

I also noticed that the to/from addresses are capitalized. 
image

In another PR previously, @Jibz1 had addressed it. So, just want to call this out in case this is a regression. Or may be that other PR never shipped. 🤔

**Comment by Jibz1** [comment]
@williamrobertson13 can you take a look at  https://github.com/MystenLabs/sui/pull/8396

File: apps/wallet/src/ui/app/components/receipt-card/index.tsx [hunk 1]
```diff
@@ -104,95 +132,119 @@ function ReceiptCard({ txn, activeAddress }: ReceiptCardProps) {
...
+                                                  <TxnAmount
+                                                      amount={amount}
+                                                      label={txnStatusText}
+                                                      coinType={coinType}
+                                                  />
+
+                                                  <TxnAddress
+                                                      address={recipientAddress}
+                                                      label={
+                                                          isSender
```
[the remark this answers is not in the document]
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/receipt-card/index.tsx:198]: Take a look at this https://github.com/MystenLabs/sui/pull/8396;

**Comment by williamrobertson13** [comment]
> Looks good in general Will. Quick question...is the "ERROR" text in this screenshot supposed to be a placeholder for the error message? image

The "ERROR" text is placeholder! I fixed the color 😄 
image

File: apps/wallet/src/ui/app/components/receipt-card/SponsoredTxnGasSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+type SponsoredTxnGasSummaryProps = {
+    totalGas: number;
+    sponsor: SuiAddress;
+};
+
+export function SponsoredTxnGasSummary({
+    totalGas,
+    sponsor,
+}: SponsoredTxnGasSummaryProps) {
+    const [senderTotalAmount, senderTotalAmountSymbol] = useFormatCoin(
```
Comment (Jibz1) [inline apps/wallet/src/ui/app/components/receipt-card/SponsoredTxnGasSummary.tsx:20]: why format 0?

File: apps/wallet/src/ui/app/components/receipt-card/index.tsx [hunk 3]
```diff
@@ -81,15 +82,42 @@ function ReceiptCard({ txn, activeAddress }: ReceiptCardProps) {
...
+    const isSender = activeAddress === transactionSender;
+    const isSponsoredTransaction = transactionSender !== owner;
+    const gasTotal = getTotalGasUsed(txn);
+
+    const showGasSummary = isSuccessful && isSender && gasTotal;
+    const showSponsorInfo = !isSuccessful && isSender && isSponsoredTransaction;
+
+    let txnGasSummary: JSX.Element | undefined;
+    if (showGasSummary && isSponsoredTransaction) {
+        txnGasSummary = (
```
Comment (Jibz1) [inline apps/wallet/src/ui/app/components/receipt-card/index.tsx:99]: I wonder if we do something like this and inline `txnGasSummary`
```suggestion
        txnGasSummary = isSponsoredTransaction ? <SponsoredTxnGasSummary sponsor={owner} totalGas={gasTotal} /> : <TxnGasSummary
                 totalGas={gasTotal}
                 transferAmount={totalSuiAmount}
             />
```

**Review by Jibz1 — APPROVED** [review]

**Review by mystie711 — APPROVED** [review]

File: apps/wallet/src/ui/app/components/receipt-card/SponsoredTxnGasSummary.tsx (hunk 2 above)
Comment (williamrobertson13) [reply apps/wallet/src/ui/app/components/receipt-card/SponsoredTxnGasSummary.tsx]: Discussed this offline - I confirmed that a sponsor always has to pay 100% of the gas for a transaction hence why the sender pays 0 SUI in this case

**Pull request merged**